plugins/common/editor/api.ts

```diff
@@ -0,0 +1,12 @@
+import { Position } from "./type";
+
+export interface EditorApi {
+  methods: {
```

Comment:
During a refactoring with @LianaHus we were merging editor and sourcehighlight and added to more functions:
https://github.com/ethereum/remix-ide/blob/master/src/app/editor/editor.js#L47
`'clearAnnotations', 'addAnnotation'` (for adding content to the gutter)
would be nice to see them here

Reply:
What is the data model of annotation ?

Reply:
here it is
```
[
 {
  "row": 19,
  "column": 5,
  "text": " text or errror ... ",
  "type": "error"
 }
]"
```

Reply:
type can be ` error || warning || information`